src/components/StatsUsageChart.vue

```diff
@@ -1,5 +1,6 @@
 <template>
   <base-line-chart
+    :key="season.id"
```

Comment:
I would love to see if this could be moved to the base chart component so any other usage benefits from it.

@itsalaidbacklife it might also be better to switch away from this library altogether and either directly use it or try some alternatives. There are a lot of reactivity problems with this one in its current iteration.

Reply:
I figured out the issue and you don't need a key to solve it-- we were missing a part yesterday when we paired on it. The issue is more about reactivity and when values update with how we're passing props than anything else. Consider the following in `src/components/StatsUsageChart.vue`:
```
<base-line-chart
  id="usage-stats-chart"
  :title="title"
  :data-sets="dataSets"
  :labels="labels"
/>
```

This is a totally valid usage and passes a few props. So then we move on to `src/components/BaseLineChart.vue`:
```
<Line
  :id="instanceId"
  :options="chartOptions"
  :data="chartData"
/>
```

This sets the id (which needs a little work because the default isn't logical and is in fact invalid, but I digress) and also sets the options and data. The data IS reactive, but the chartOptions are not-- why? Lets look at how they're defined:

```
computed: {
  chartData() {
    return {
      labels: this.labels,
      datasets: this.dataSets,
    };
  },
},
```

We have a computed props for the options and those update every single time. Computed props will properly update when a change to the prop is detected. This is reflected in the chart when the lines change as the seasons are changed.

The chartOptions don't update though, and how are they defined?

```
data() {
  return {
    chartOptions: {
      responsive: true,
      plugins: {
        title: {
          text: this.title,
          display: !!this.title,
        },
      },
    },
  };
},
```

At first glance, this looks right-- they're using local component state and as we all know local component state is reactive _when the value changes_. The issue is that the prop for title is changing, but it's never actually re-setting the local state via a setter. It's just changing and leaving the state alone. You need to actually convert this object to a computed prop in order for reactivity to start working again. This means you don't need the `:key` prop and are able to benefit from the reactivity for performance reasons.

-----

In `BaseLineChart.vue` you should also convert the id to use a new static property on the `created` hook to use the unique id of the component when not set so you benefit from a truly unique id if it's not set.

On the element:
```
:id="instanceId"
```
Prop validation:
```
props: {
  id: {
    type: String,
    required: false,
  },
```
Static property:
```
created() {
  this.instanceId = this.id || this._uid;
},
```

Let me know if you have any questions -- and nice work. This was a tricky one!

Reply:
Hey Sean, thanks for taking a look into this and helping resolve it. I was able to put in your suggestion and get it pushed up.

After testing it looks like it works perfect. Let me know if any other changes need to be made.